5-cluster-implementation.md

```diff
@@ -13,20 +13,20 @@
 
 ## Background
 
-When designing the `DockerCluster` api kind we need to make sure that we adhere to the contract for a [Cluster Infrastructure Provider](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html) and expose enough information to the user so that we can create any required infrastructure for the cluster (but not machines for nodes).
+When designing the `DockerCluster` API kind we need to make sure that we adhere to the contract for a [Cluster Infrastructure Provider](https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure.html) and expose enough information to the user so that we can create any required infrastructure for the cluster (but not machines for nodes).
 
-In our example the infrastructure we need to provision is a load balancer to sit in front of our the api servers of our control plane nodes (that will be provisioned later). To do this we will create an instances of a HAProxy container. The address of the load balancer will be used in the `controlPlaneEndPoint`.
+In our example, the infrastructure that we need to provision is a load balancer to sit in front of our the API servers of our control plane nodes (that will be provisioned later). To do this, we will create an instance of an HAProxy container. The address of the load balancer will be specified in `controlPlaneEndPoint`.
 
 When implementing the controller for `DockerCluster` (and `DockerMachine`) you will generally follow a pattern similar to this:
 
-1. In **Reconcile** get the instance of the API type being reconciled
+1. In **Reconcile**, get the instance of the API type being reconciled
 2. Get the owning CAPI type (i.e. if we are reconciling **DockerCluster** then we get **Cluster**)
    1. You may need to other types if you are reconciling machines
-3. If we don't have the owning CAPI type (because the "owner reference isn't set yet") then exit
-4. If the instance has a non-zero deletion timestamp, then call a **reconcileDelete** function that does:
-   1. do any actions to delete the infrastructure
-   2. remove the finalizer & save/patch
-5. If the has a zero deletion timestamp, then call a **reconcileNormal** function that does:
+3. If the owning CAPI type doesn't exist yet (because the "owner reference isn't set yet"), then exit
+4. If the instance has a non-zero deletion timestamp (indicating that it has been marked for deletion), then call a **reconcileDelete** function that:
+   1. performs any actions to delete the infrastructure
```

Comment:
we can perhaps say, do any cleanup activities and then remove the finalizers. Only after you remove the finalizers, you will be able to delete the underlying resource.

Reply:
I'm not sure I follow - the text already says that?  1. cleanup, 2. remove the finalizer & save/patch